src/data_provider/src/sysInfoSolaris.cpp

```diff
@@ -197,10 +200,82 @@ nlohmann::json SysInfo::getNetworks() const
 
     return networks;
 }
+
 nlohmann::json SysInfo::getPorts() const
 {
-    return nlohmann::json();
+    nlohmann::json ports{};
+    std::vector<mib_item_t> items;
+
+    // Open /dev/arp and push into ioctl()
+    int sd = PortSolarisHelper::mibOpen();
+
+    if (sd != -1)
+    {
+        mib2_tcpConnEntry_t* tp;
+        mib2_tcp6ConnEntry_t* tp6;
+        mib2_udpEntry_t* ud;
+        mib2_udp6Entry_t* ud6;
+
+        // Get all connections available
+        PortSolarisHelper::mibGetItems(sd, items);
+
+        // Extract constant sizes - need do once only
+        PortSolarisHelper::mibGetConstants(items);
+
+        for (auto item : items)
+        {
+            if (item.group == MIB2_TCP)
+            {
+                for (tp = (mib2_tcpConnEntry_t*)item.val.get();
```

Comment:
Here we have four times the same code, and we only change the structure we are using.
Have you considered a C++ trick to avoid this? For example, `std::variant`

Reply:
making use of this alternative makes the code excessively complex, which is originally intended to be C